Directory.Build.targets

```diff
@@ -105,9 +105,8 @@
     <Compile Include="$(SharedSourceRoot)ReferenceAssemblyInfo.cs" LinkBase="Properties" />
   </ItemGroup>
 
-  <PropertyGroup Condition="'$(Language)' == 'C#'">
+  <PropertyGroup Condition="'$(Language)' == 'C#' AND '$(IsReferenceAssemblyProject)' == 'true'">
```

Comment:
@JunTaoLuo and @wtgodbe if this PR doesn't work, will need to make this change in #14425. Implementation assemblies in that PR have incorrect versions.

Reply:
Ah I see.

Reply:
Please resolve this comment when #14425 is fixed up

Reply:
Oh wait, I did make this fix already. Otherwise we wouldn't run into the assembly version mismatch between implementation and ref assemblies.